javalib/src/main/scala/java/util/package.scala

```diff
@@ -29,7 +31,194 @@ package object util {
   private[util] def defaultOrdering[E]: Ordering[E] = {
     new Ordering[E] {
       def compare(a: E, b: E): Int =
-        a.asInstanceOf[Comparable[E]].compareTo(b)
+        if (a === null) {
+          if (b === null) 0
+          else -1
+        } else {
+          a.asInstanceOf[Comparable[E]].compareTo(b)
+        }
```

Comment:
It allows `null` but *not* with the *natural ordering*. I am pretty sure, at least. It only allows `null` if you give it a comparator that supports nulls.